### PR TITLE
add interceptor to check transaction ready to generate filing

### PR DESCRIPTION
**Pull request by ICoxCH**

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/FilingInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@Component
+public class FilingInterceptor implements HandlerInterceptor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FilingInterceptor.class);
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws IOException {
+        final var transaction = (Transaction) request.getAttribute("transaction");
+
+        if (transaction == null) {
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/confirmationstatementapi/interceptor/FilingInterceptor.java:23]: think it just needs a test to cover transaction == null

**Review by markpit — APPROVED** [review]
